library/trans.c

```diff
@@ -766,7 +785,15 @@ static void dnet_trans_enqueue_responses_on_timed_out(struct dnet_node *n, struc
 	list_for_each_entry_safe(r, tmp, timeout_responses, req_entry) {
 		list_del_init(&r->req_entry);
 
-		dnet_schedule_io(n, r);
+		if (r->io_req_type == DNET_IO_REQ_TYPED_RESPONSE) {
+			n2_response_info_call_response(r->response_info);
```

Comment:
Why is it called here, not from io pool?

Reply:
This call creates another `dnet_io_req` and place it into io pool queue - temporary hack

Reply:
(was discussed by voice)
Repliers (that are source for `response_info`) are created respecting to desired behavior: https://github.com/interiorem/elliptics/blob/2ae768f7ef1b527998bb6834d7008a63c023118d/library/net.cpp#L1442